tests/pocket_test.cpp

```diff
@@ -1330,6 +1344,95 @@ TEST_CASE( "character best pocket", "[pocket][character][best]" )
     // Includes worn items
     // Uses best of multiple worn items
     // Skips avoided items
+
+    avatar &they = get_avatar();
+    map &here = get_map();
+    clear_avatar();
+    clear_map();
+
+    // Regression test for #52422:
+    // Wearing only a backpack containing a rope,
+    // When picking up and M4 from the ground,
+    // the M4 should go into the backpack, not into the rope
+
+    // Spawn items on the map at this location
+    item m4a1 = here.add_item( they.pos(), item( "m4_carbine" ) );
+    item rope = here.add_item( they.pos(), item( "rope_6" ) );
+    item backpack = here.add_item( they.pos(), item( "backpack_hiking" ) );
+    // Rope and backpack are containers, both capable of holding the M4
+    REQUIRE( rope.is_container() );
+    REQUIRE( backpack.is_container() );
+    REQUIRE( rope.can_contain( m4a1 ).success() );
+    REQUIRE( backpack.can_contain( m4a1 ).success() );
+    REQUIRE( backpack.can_contain( rope ).success() );
+
+    // Wearing the backpack, the best pocket for the m4 should be the backpack
+
+    GIVEN( "avatar is wearing only a hiking backpack" ) {
+        they.worn.clear();
+        they.wear_item( backpack );
```

Comment:
Wearing also creates a new reference, but thankfully returns it (as an optional list iterator, heh). So you need to replace the backpack in the following checks with that. That will make the `i_add` tests pass.

Reply:
Thanks, using the doubly-dereferenced returned iterator gets me the backpack clone I'm looking for, and allows subsequent steps to pass.